app/Bots/IRCAnnounceBotExternal.php

```diff
@@ -0,0 +1,207 @@
+            $originEnum = 1;
+        }
+
+        if ($torrent->personal_release) {
+            $originEnum = 2;
+        }
+
+        $leechTypeEnum = 0;
+
+        if ($torrent->free > 0) {
+            switch ($torrent->free) {
+                case 100:
+                    $leechTypeEnum = 1;
+
+                    break;
+                case 75:
+                    $leechTypeEnum = 2;
+
+                    break;
+                case 50:
+                    $leechTypeEnum = 3;
+
+                    break;
+                case 25:
+                    $leechTypeEnum = 4;
+
+                    break;
+            }
+        }
+
+        if ($torrent->doubleup) {
+            switch ($leechTypeEnum) {
+                case 0:
+                    $leechTypeEnum = 5;
+
+                    break;
+                case 1:
+                    $leechTypeEnum = 6;
+
+                    break;
+                case 2:
+                    $leechTypeEnum = 7;
+
+                    break;
+                case 3:
+                    $leechTypeEnum = 8;
+
+                    break;
+                case 4:
+                    $leechTypeEnum = 9;
+            }
+        }
+
+        $meta = null;
+        $category = $torrent->category;
+
+        if ($torrent->tmdb_movie_id > 0 || $torrent->tmdb_tv_id > 0) {
+            $meta = match (true) {
+                $category->tv_meta    => TmdbTv::find($torrent->tmdb_tv_id),
+                $category->movie_meta => TmdbMovie::find($torrent->tmdb_movie_id),
+                default               => null,
+            };
+        }
+
+        return self::post([
+            'id'                     => (string) $torrent->id,
+            'url'                    => \sprintf('%s/torrents/%d', $appurl, $torrent->id),
+            'name'                   => $torrent->name,
+            'uploader'               => $torrent->anon ? 'Anonymous' : $torrent->user->username,
+            'size'                   => $torrent->getSize(),
+            'size_bytes'             => $torrent->size,
+            'announce_type_enum'     => $announceTypeEnum,
+            'category_enum'          => $category->id,
+            'category_name'          => $category->name,
+            'origin_enum'            => $originEnum,
+            'leech_type_enum'        => $leechTypeEnum,
+            'upload_time_unix_epoch' => $torrent->created_at->getTimestamp(),
+            'freeleech'              => (bool) $torrent->free > 0,
+            'freeleech_percent'      => $torrent->free,
+            'double_up'              => $torrent->doubleup,
+            'resolution'             => $torrent->resolution?->name ?? '',
+            'type'                   => $torrent->type->name,
+            'release_year'           => isset($torrent->meta->release_date) ? $torrent->meta->release_date->format('Y') : (isset($torrent->meta->first_air_date) ? $torrent->meta->first_air_date->format('Y') : null),
```

Comment:
It looks like you intended to use the `$meta` variable here instead of `$torrent->meta`.